src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -191,6 +238,12 @@ impl EventLoop {
         timeout: Option<Duration>,
         mut app: A,
     ) -> PumpStatus {
+        if let Some(pump_event_notifier) = self.pump_event_notifier.as_ref() {
+            // Notify that we should start waiting, since we're in winit.
+            *pump_event_notifier.control.0.lock().unwrap() = true;
+            pump_event_notifier.control.1.notify_one();
+        }
```

Comment:
```suggestion
```

Reply:
yeah, that is redundant, though, I left for consistency.

Reply:
You shouldn't wake up the thread before having dispatched all messages since it will just immediately trigger a notification again.

Reply:
it would have been blocked in `read_events`, otherwise we'd had events to read and thus wake-up would have been needed anyway. But yeah should be fine.